src/graphql/client_graph_node.rs

```diff
@@ -44,11 +57,97 @@ pub async fn query_graph_node_network_block_hash(
     if let Some(data) = response_body.data {
         match data.block_hash_from_number {
             Some(hash) => Ok(hash),
-            None => Err(QueryError::EmptyResponseError(
+            None => Err(QueryError::ParseResponseError(
                 "No block hash from number".to_string(),
             )),
         }
     } else {
-        Err(QueryError::EmptyResponseError(graph_node_endpoint))
+        Err(QueryError::ParseResponseError(format!(
+            "Network {network} blockhash at block {block_number}"
+        )))
     }
 }
+
+/// Query graph node for Indexing Statuses
+pub async fn perform_indexing_statuses(
+    graph_node_endpoint: String,
+    variables: indexing_statuses::Variables,
+) -> Result<reqwest::Response, reqwest::Error> {
+    let request_body = IndexingStatuses::build_query(variables);
+    let client = reqwest::Client::new();
+    client
+        .post(graph_node_endpoint)
+        .json(&request_body)
+        .send()
+        .await?
+        .error_for_status()
+}
+
+/// Construct GraphQL variables and parse result for Proof of Indexing.
+/// For other radio use cases, provide a function that returns a string
```

Comment:
can you elaborate on this comment 👀

Reply:
oh! outdated from the poi radio, will update

Reply:
updated to a more refactored version with more comments so this long function is now in 3 smaller funcs. It is a bit more expensive now in terms of querying the data twice due to ownership being moved around + not trivial to implement Copy, but will take suggestions!